WPThemeReview/Tests/CoreFunctionality/PrefixAllGlobalsUnitTest.php

```diff
@@ -0,0 +1,84 @@
+<?php
+/**
+ * Unit test class for WPThemeReview Coding Standard.
+ *
+ * @package WPTRT\WPThemeReview
+ * @link    https://github.com/WPTRT/WPThemeReview
+ * @license https://opensource.org/licenses/MIT MIT
+ */
+
+namespace WPThemeReview\Tests\CoreFunctionality;
+
+use PHP_CodeSniffer\Tests\Standards\AbstractSniffUnitTest;
+
+/**
+ * Unit test class for the PrefixAllGlobals sniff.
+ *
+ * @package WPTRT\WPThemeReview
+ *
+ * @since 0.2.0
+ */
+class PrefixAllGlobalsUnitTest extends AbstractSniffUnitTest {
+
+	/**
+	 * Error files with the expected nr of errors.
+	 *
+	 * @var array
+	 */
+	private $expected_results = array(
+
+		/*
+		 * In /CoreFunctionality/ThemeTemplatesException.
+		 */
+
+		// Files in a theme context.
+		'footer_widgets.inc' => 1,
+		'header.inc'         => 1,
+	);
+
+	/**
+	 * Get a list of all test files to check.
+	 *
+	 * @param string $testFileBase The base path that the unit tests files will have.
+	 *
+	 * @return string[]
+	 */
+	protected function getTestFiles( $testFileBase ) {
+
+		$sep        = \DIRECTORY_SEPARATOR;
+		$test_files = glob( dirname( $testFileBase ) . $sep . 'ThemeTemplatesException{' . $sep . ',' . $sep . '*' . $sep . '}*.inc', \GLOB_BRACE );
+
+		if ( ! empty( $test_files ) ) {
+			return $test_files;
+		}
+
+		return array( $testFileBase . '.inc' );
+	}
+
+	/**
+	 * Returns the lines where errors should occur.
+	 *
+	 * @param string $testFile The name of the file being tested.
+	 * @return array <int line number> => <int number of errors>
+	 */
+	public function getErrorList( $testFile = '' ) {
+
+		if ( isset( $this->expected_results[ $testFile ] ) ) {
```

Comment:
See my remarks above. Only the test for the non-template file should trigger an error, none of the others should.

Reply:
So I should remove this. But how do I test for the specific `.inc` files if they are triggering errors? Like `footer_widgets.inc` and say `social-share.inc` (outside of `partials` folder) that should also trigger an error.

Reply:
The WPCS file name sniff is the only one to include a list like that in a property as all the errors would be on the same line and it improved the documentation of why and what.

Normally for sniffs which need to be tested using several files, the test method would use a unit test file name based `switch` which also allows more easily to the differentiate between the lines on which errors/warnings are being thrown between files.
The `default` case would be an empty array (explicitly expect no errors), so you would only need to list the non-template test case files in the `switch`.

Like in the file you linked above as well as in, for instance, this one: https://github.com/WPTRT/WPThemeReview/blob/develop/WPThemeReview/Tests/ThouShallNotUse/NoAutoGenerateUnitTest.php

> Should I manually include the test files,

'The `switch` doesn't include the test files. Test case files with a `SniffNameTest.1.inc` (numbered) file name will be included for testing automatically by PHPCS, it's only for non-standard test case file names (which you need for this sniff) that you have to tell PHPCS which files to use in the test and you already do so via the `getTestFiles()` method.